examples/material_gallery/lib/demo/markdown_demo.dart

```diff
@@ -0,0 +1,46 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_markdown/flutter_markdown.dart';
+
+class MarkdownDemo extends StatefulComponent {
+  MarkdownDemo({Key key}) : super(key: key);
+
+  _MarkdownDemoState createState() => new _MarkdownDemoState();
+}
+
+class _MarkdownDemoState extends State<MarkdownDemo> {
+
+  void initState() {
+    super.initState();
+
+    DefaultAssetBundle.of(context).loadString('packages/flutter_gallery_assets/markdown.md').then((String fileContents) {
+      setState(() {
+        
+        markdownSrc = fileContents;
+      });
+    });
+  }
+
+  String markdownSrc;
+
+  Widget build(BuildContext context) {
+    if (markdownSrc == null) {
+      return new Scaffold(
+        toolBar: new ToolBar(center: new Text("Markdown"))
+      );
+    }
+
+    return new Scaffold(
+      toolBar: new ToolBar(center: new Text("Markdown")),
+      body: new Block(
+        padding: new EdgeDims.all(16.0),
+        children: <Widget>[
+          new Markdown(data: markdownSrc)
```

Comment:
Rather than handing the String directly to the Markdown widget, we should create an immutable MarkdownData object that holds the string and can cache data structures related to parsing the markdown string.